ci/build_cpp.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -e
+
+################################################################################
+# SETUP - Check environment
+################################################################################
+
+gpuci_logger "Get env"
+env
+
+gpuci_logger "Activate conda env"
+. /opt/conda/etc/profile.d/conda.sh
+conda activate rapids
+
+gpuci_logger "Check versions"
+python --version
+$CC --version
+$CXX --version
+
+gpuci_logger "Check conda environment"
+conda info
+conda config --show-sources
+conda list --show-channel-urls
```

Comment:
It seems like these `Check environment` steps exist in 3 places currently. To keep everything DRY, let's move these to their own script, `check_environment.sh`, which we can invoke with `source check_environment.sh`.